core/client/fs/src/main/java/alluxio/client/file/DoraCacheFileSystem.java

```diff
@@ -84,6 +88,11 @@ public URIStatus getStatus(AlluxioURI path, GetStatusPOptions options)
 
       return mDoraClient.getStatus(ufsFullPath.toString(), mergedOptions);
     } catch (RuntimeException ex) {
+      if (ex instanceof StatusRuntimeException) {
+        if (((StatusRuntimeException) ex).getStatus().equals(Status.NOT_FOUND)) {
+          throw new FileNotFoundException();
+        }
+      }
```

Comment:
Inspecting the runtime type of exceptions can get very cumbersome. Instead of catching all `RuntimeException`s here, I think it's the responsibility of `DoraCacheClient` to do convert the `io.grpc.StatusRuntimeException` to an Alluxio exception with semantics (e.g. `FileDoesNotExistException`). So that `DoraCacheClientFileSystem` can catch more semantic exceptions.

Reply:
Maybe we should have some new API:
```
AlluxioRuntimeException#fromGrpcStatusRuntimeException(StatusRuntimeException e)
```

But we still need to check if this exception is from GRPC or from other places.
The same logic as here is needed to check if a `AlluxioRuntimeException` is `FileDoesNotExistException`.
Only when we have some specific exceptions (e.g. FileDoesNotExistException), we will throw the exceptions and return. Otherwise, we have to talk to UFS.

@jja725 has a plan to do better exception handlings on server and client sides.